modules/components/org.wso2.carbon.appfactory.resource.mgt/src/main/java/org/wso2/carbon/appfactory/resource/mgt/listeners/CloudEnvironmentPermissionListener.java

```diff
@@ -0,0 +1,138 @@
+import org.wso2.carbon.appfactory.resource.mgt.internal.ServiceHolder;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.stratos.common.beans.TenantInfoBean;
+import org.wso2.carbon.stratos.common.exception.StratosException;
+import org.wso2.carbon.stratos.common.listeners.TenantMgtListener;
+import org.wso2.carbon.user.api.AuthorizationManager;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.api.UserStoreManager;
+
+import java.util.Set;
+
+public class CloudEnvironmentPermissionListener implements TenantMgtListener {
+	private static final Log log = LogFactory.getLog(CloudEnvironmentPermissionListener.class);
+	private static final int EXEC_ORDER = 11;
+
+	@Override
+	public int getListenerOrder() {
+		// TODO Auto-generated method stub
+		return EXEC_ORDER;
+	}
+
+	@Override
+	public void onSubscriptionPlanChange(int arg0, String arg1, String arg2)
+			throws StratosException {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public void onTenantActivation(int arg0) throws StratosException {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public void onTenantCreate(TenantInfoBean tenInfoBean) throws StratosException {
+		log.info("*********adding permissions******") ;
+		int tenantId = tenInfoBean.getTenantId();
+		PrivilegedCarbonContext.startTenantFlow();
+		PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId, true);
+
+		String currentCloudStage = System.getProperty(AppFactoryConstants.CLOUD_STAGE);
+		if (currentCloudStage == null || currentCloudStage.isEmpty()) {
+			log.error(AppFactoryConstants.CLOUD_STAGE + " system variable is not set. No permissions related to cloud environments are added.");
+			throw new StratosException(AppFactoryConstants.CLOUD_STAGE + " system variable is not set. No permissions related to cloud environments are added.");
+		}
+		try {
+			addCloudRolePermissions(tenInfoBean, currentCloudStage);
+
+		} catch (UserStoreException e) {
+			String msg = "Error while authorizing permissions defined in appfactory.xml";
+			log.error(msg, e);
+
+		} catch (AppFactoryException e) {
+			String msg = "Error while authorizing permissions defined in appfactory.xml";
+			log.error(msg, e);
+
+		} finally {
+
+			PrivilegedCarbonContext.endTenantFlow();
+
+		}
+
+	}
+
+	private void addCloudRolePermissions(TenantInfoBean tenantInfoBean, String currentCloudStage) throws AppFactoryException, UserStoreException {
+		String adminUser = tenantInfoBean.getAdmin();
+		String rolePermissionConfigPath = AppFactoryConstants.DEPLOYMENT_STAGES + "." + currentCloudStage + "." + AppFactoryConstants.TENANT_ROLES_ROLE;
+		Set<RoleBean> roleBeanList;
+		try {
+			roleBeanList = AppFactoryUtil.getRolePermissionConfigurations(rolePermissionConfigPath, adminUser);
+		} catch (AppFactoryException e) {
+			String errorMsg = "Failed to get permission collections from appfactory configuration.";
+			log.error(errorMsg, e);
```

Comment:
since this is a private method, do we really need to log the error?. You are already logging it in the calling method